src/DIRAC/Core/Utilities/TimeUtilities.py

```diff
@@ -73,19 +75,25 @@ def timed(*args, **kw):
                     except IndexError:
                         argsLen = ""
 
-        print("%s Exec time ===> function %r %s -> %2.2f sec" % (pre, method.__name__, argsLen, te - ts))
+        gLogger.info("%s Exec time ===> function %r %s -> %2.2f sec" % (pre, method.__name__, argsLen, te - ts))
         return result
 
     return timed
 
 
 def toEpoch(dateTimeObject=None):
     """
-    Get seconds since epoch
+    Get seconds since epoch. Accepts datetime or date objects
     """
-    if not dateTimeObject:
-        dateTimeObject = datetime.datetime.utcnow()
-    return calendar.timegm(dateTimeObject.utctimetuple())
+    try:
+        # If object of date type is given, return a datetime object set to midnight
+        if type(dateTimeObject) is datetime.date:
```

Comment:
```suggestion
        if isinstance(dateTimeObject, datetime.date):
```

Reply:
This unfortunately does not work as a datetime.date type it returns `True` for both `isinstance(dateTimeObject, datetime.date)` and `isinstance(dateTimeObject, datetime.datetime)`

Reply:
@chrisburr mainly we want to double check that the `dateTypeObject` is not a `datetime.date` type, so that would still return `True` for both date and datetime objects no?

Reply:
Yes we want that specific condition for only datetime.date objects. That if statement doesnt filter them out.
```python
In [19]: dateTimeObject = datetime.datetime.utcnow()

In [20]: if not isinstance(dateTimeObject, datetime.time):
    ...:     print("Type: ", type(dateTimeObject))
    ...:     print("True")
    ...: 
Type:  <class 'datetime.datetime'>
True

In [21]: dateTimeObject = datetime.date.today()

In [22]: if not isinstance(dateTimeObject, datetime.time):
    ...:     print("Type: ", type(dateTimeObject))
    ...:     print("True")
    ...: 
Type:  <class 'datetime.date'>
True
```

Reply:
Ah I see the problem now, let me have a think...

Reply:
How about:
```python
def toEpoch(dateTimeObject=None):
    """
    Get seconds since epoch. Accepts datetime or date objects
    """
    return toEpochMilliSeconds(dateTimeObject) // 1000


def toEpochMilliSeconds(dateTimeObject=None):
    """
    Get milliseconds since epoch
    """
    # Is this supposed to handle dateTimeObject == None? Else why does it have a default value?
    # If it should then you need:
    # if dateTimeObject == None:
    #     dateTimeObject = datetime.datetime.now()
    if dateTimeObject.resolution == datetime.timedelta(days=1):
        # Add time information corresponding to midnight UTC if it's a datetime.date
        dateTimeObject = datetime.datetime.combine(dateTimeObject, datetime.time.min.replace(tzinfo=datetime.timezone.utc))
    posixTime = dateTimeObject.replace(tzinfo=datetime.timezone.utc).timestamp()
    return int(posixTime * 1000)
```

Reply:
A few further comments:
* `toEpoch` should use `toEpochMilliSeconds` rather than the other way around to avoid loosing precision and rounding to the nearest 1000 milliseconds
* The `dateTimeObject == None` should either be handled or reoved from the signature
* Does this need to handle timezone naive datetimes? Or just timezone aware ones? Either way it should be tested.